core/templates/dev/head/components/question-directives/questions-list/questions-list.directive.ts

```diff
@@ -190,43 +203,75 @@ oppia.directive('questionsList', [
               controller: [
                 '$scope', '$uibModalInstance',
                 function($scope, $uibModalInstance) {
-                  $scope.selectedSkillIds = [];
-                  $scope.skillSummaries = allSkillSummaries;
-                  $scope.skillSummaries.forEach(function(summary) {
-                    summary.isSelected = false;
-                  });
+                  var init = function() {
+                    $scope.currentMode = currentMode;
+                    $scope.linkedSkillsWithDifficulty =
+                      linkedSkillsWithDifficulty;
+                    $scope.skillSummaries = allSkillSummaries;
+                  };
 
                   $scope.selectOrDeselectSkill = function(skillId, index) {
                     if (!$scope.skillSummaries[index].isSelected) {
-                      $scope.selectedSkillIds.push(skillId);
+                      $scope.linkedSkillsWithDifficulty.push(
+                        SkillDifficultyObjectFactory.create(
+                          skillId,
+                          $scope.skillSummaries[index].getDescription(),
+                          DEFAULT_SKILL_DIFFICULTY));
                       $scope.skillSummaries[index].isSelected = true;
                     } else {
-                      var idIndex = $scope.selectedSkillIds.indexOf(skillId);
-                      $scope.selectedSkillIds.splice(idIndex, 1);
+                      var idIndex = $scope.linkedSkillsWithDifficulty.map(
+                        function(linkedSkillWithDifficulty) {
+                          return linkedSkillWithDifficulty._id;
+                        }).indexOf(skillId);
+                      $scope.linkedSkillsWithDifficulty.splice(idIndex, 1);
                       $scope.skillSummaries[index].isSelected = false;
                     }
                   };
 
-                  $scope.done = function() {
-                    $uibModalInstance.close($scope.selectedSkillIds);
+                  $scope.goToSelectSkillView = function() {
+                    $scope.currentMode = MODE_SELECT_SKILL;
                   };
 
-                  $scope.cancel = function() {
+                  $scope.goToSelectDifficultyView = function() {
+                    $scope.currentMode = MODE_SELECT_DIFFICULTY;
+                  };
+
+                  $scope.startQuestionCreation = function() {
+                    var skillIds = [];
+                    var skillDifficulties = [];
+                    $scope.linkedSkillsWithDifficulty.forEach(
+                      function(skillWithDifficulty) {
+                        skillIds.push(skillWithDifficulty._id);
+                        skillDifficulties.push(skillWithDifficulty._difficulty);
+                      });
+                    var skillIdsAndDifficulties = {
```

Comment:
Can you inline this in line 251? (I.e. don't give it a name)

That's because this is just supposed to be a "dict passed out of modal" rather than a conceptual object on its own (otherwise we should be using a domain object for the latter).

Reply:
Also, I might be missing something, but perhaps you don't even need to split it out into two lists at all. Couldn't you just pass in a copy of linkedSkillsWithDifficulty?

Reply:
I am passing it as is, but I do need to separate it out later for question creation, so I have done that in the .then() function